be/src/util/byte_buffer.h

```diff
@@ -52,6 +52,13 @@ struct ByteBuffer {
         return ptr;
     }
 
+    static ByteBufferPtr reallocate(const ByteBufferPtr& old_ptr, size_t new_size) {
+        ByteBufferPtr ptr(new ByteBuffer(new_size));
```

Comment:
Reallocate:  only support alloc larger.
Flip: make smaller and free up excess memory

Reply:
Updated. It will do nothing when the new_size is less than the origin capacity.